app/assets/javascripts/components/contextual-guidance.js

```diff
@@ -0,0 +1,69 @@
+function ContextualGuidance ($module) {
+  if (!$module) $module = document
+  this.$module = $module
+  this.$fields = $module.querySelectorAll('[data-contextual-guidance]')
+}
+
+function nodeListForEach (nodes, callback) {
+  if (window.NodeList.prototype.forEach) {
+    return nodes.forEach(callback)
+  }
+  for (var i = 0; i < nodes.length; i++) {
+    callback.call(window, nodes[i], i, nodes)
+  }
+}
+
+ContextualGuidance.prototype.handleFocus = function (event) {
+  // Get the target element
+  var target = event.target
+  var guidanceId = target.getAttribute('data-contextual-guidance')
```

Comment:
Could we refactor uses of guidanceId into a method that returns a guidance, given a target? I notice the same getAttribute/querySelector code is being repeated here.